components/script/fetch.rs

```diff
@@ -159,32 +179,38 @@ pub(crate) fn Fetch(
 
     // Step 2. Let requestObject be the result of invoking the initial value of Request as constructor
     //         with input and init as arguments. If this throws an exception, reject p with it and return p.
-    let request = match Request::Constructor(global, None, can_gc, input, init) {
+    let request_object = match Request::Constructor(global, None, can_gc, input, init) {
         Err(e) => {
             response.error_stream(e.clone(), can_gc);
             promise.reject_error(e, can_gc);
             return promise;
         },
-        Ok(r) => {
-            // Step 3. Let request be requestObject’s request.
-            r.get_request()
-        },
+        Ok(r) => r,
     };
+    // Step 3. Let request be requestObject’s request.
+    let request = request_object.get_request();
     let timing_type = request.timing_type();
 
-    let mut request_init = request_init_from_request(request);
-    request_init.policy_container =
-        RequestPolicyContainer::PolicyContainer(global.policy_container());
-
     // Step 4. If requestObject’s signal is aborted, then:
-    // TODO
-    // Step 4.1. Abort the fetch() call with p, request, null, and requestObject’s signal’s abort reason.
-    // TODO
-    // Step 4.2. Return p.
-    // TODO
+    if let Some(signal) = request_object.GetSignal() {
+        if signal.aborted() {
+            // Step 4.1. Abort the fetch() call with p, request, null, and requestObject’s signal’s abort reason.
+
+            // TODO: Figure out how to do this:
+            // rooted!(in(*cx) let mut abort_reason = UndefinedValue());
+            // signal.Reason(cx, abort_reason.handle_mut());
```

Comment:
How do I do this? The `abort_reason` can be anything, but `promise_reject` needs to take an `Error`. For now I set it to `Error::Abort` which passes most tests, but that's not fully correct.

Reply:
🤦